crates/flux-refineck/src/checker.rs

```diff
@@ -1430,7 +1458,10 @@ impl Mode for RefineMode {
         target: BasicBlock,
     ) -> Result<bool> {
         let bb_env = &ck.inherited.mode.bb_envs[&ck.def_id][&target];
-        debug_assert_eq!(&ck.snapshot_at_dominator(target).scope().unwrap(), bb_env.scope());
+        debug_assert_eq!(
+            ck.snapshot_at_dominator(target).scope().unwrap().bindings(),
+            bb_env.scope().bindings()
+        );
```

Comment:
was this failing? The reftgenerics should be the same throughout the entire function so the default equality implementation on `Scope` should work.

Reply:
Hmm good question, I have reverted it -- there was something going on with loops that was causing a crash but I can't reproduce it now!